test/utils/settings/data.test.js

```diff
@@ -0,0 +1,183 @@
+      .and.contain({ start: '6:00 am', rate: '0.900' })
+      .and.contain({ start: '5:30 pm', rate: '0.850' })
+      .and.contain({ start: '12:00 am', rate: '0.750' })
+      .and.contain({ start: 'Total', rate: '20.725' });
+    });
+    it('should cope with empty shedules', () => {
+      expect(
+        data.processBasalRateData(
+          multirateSettingsData.basalSchedules[1],
+        )
+      )
+      .to.have.length(1)
+      .to.contain({ start: '-', rate: '-' });
+    });
+  });
+  describe('processTimedSettings', () => {
+    it('should return formatted objects', () => {
+      expect(
+        data.processTimedSettings(
+          timedSettingsData,
+          { name: 'Sick', position: 1 },
+          timedSettingsData.units.bg,
+        )
+      )
+      .to.have.length(5)
+      .to.contain({
+        start: '12:00 am',
+        rate: '0.350',
+        bgTarget: '5.3',
+        carbRatio: 7,
+        insulinSensitivity: '2.6',
+      })
+      .and.contain({
+        start: '3:30 am',
+        rate: '0.225',
+        bgTarget: '5.0',
+        carbRatio: 10,
+        insulinSensitivity: '4.5',
+      })
+      .and.contain({
+        start: '10:00 am',
+        rate: '1.075',
+        bgTarget: '5.0',
+        carbRatio: 10,
+        insulinSensitivity: '4.5',
+      })
+      .and.contain({
+        start: '8:00 pm',
+        rate: '0.625',
+        bgTarget: '5.0',
+        carbRatio: 9,
+        insulinSensitivity: '4.5',
+      })
+      .and.contain({
+        start: 'Total',
+        rate: '15.938',
+        bgTarget: '',
+        carbRatio: '',
+        insulinSensitivity: '',
+      });
+    });
+  });
+  describe('getScheduleLabel', () => {
+    it('should return the formatted time', () => {
+      expect(
+        data.getScheduleLabel('one', 'two')
+      ).to.equal('one');
+    });
+    it('should return the formatted time', () => {
+      expect(
+        data.getScheduleLabel('one', 'one')
+      ).to.equal('one (Active at upload)');
+    });
+  });
+  describe('getTimedSchedules', () => {
+    it('should return the timed settings schedule names', () => {
+      expect(
+        data.getTimedSchedules(settingsData.basalSchedules)
+      )
+      .to.have.length(2)
+      .to.contain({ name: 'Normal', position: 0 })
+      .and.contain({ name: 'Sick', position: 1 });
+    });
+  });
+  describe('getTotalBasalRates', () => {
+    it('should return the rate total for multi rate settings', () => {
+      expect(
+        data.getTotalBasalRates(
+          multirateSettingsData.basalSchedules[0].value,
+        )
+      ).to.equal('20.725');
+    });
+  });
+  describe('getDeviceMeta', () => {
+    const timePrefs = {
+      timezoneAware: false,
+      timezoneName: null,
+    };
+    it('should return the name, schedule and date uploaded device', () => {
+      expect(
+        data.getDeviceMeta(settingsData, timePrefs)
+      ).to.have.property('name').equal('DevId0987654321');
```

Comment:
This test should be failing because you shouldn't be displaying the `deviceId` as the serial number. Looks like @krystophv did the work to make sure serial number is available but it hasn't been surfaced in the rendering.